ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift

```diff
@@ -509,4 +563,35 @@ private extension CommonFunctionality {
         }
     }
 
+    static func processRefundRequestResultWithCompletion(
+        refundRequestResult: Result<RefundRequestStatus, PublicError>,
+        completion: @escaping (ErrorContainer?) -> Void
+    ) {
+        switch refundRequestResult {
+        case let .success(refundRequestStatus):
+            switch refundRequestStatus {
+            case .success:
+                completion(nil)
+            case .userCancelled:
+                completion(Self.refundRequestError(description: "User cancelled refund request.", userCancelled: true))
+            case .error:
+                completion(Self.refundRequestError(description: "Error during refund request."))
+            }
+        case let .failure(error):
+            completion(ErrorContainer(error: error, extraPayload: [:]))
+        }
+    }
+
+    static func refundRequestError(description: String, userCancelled: Bool? = nil) -> ErrorContainer {
+        var extraPayload: [String: Any] = [:]
+        if let userCancelled = userCancelled {
+            extraPayload["userCancelled"] = userCancelled
```

Comment:
I copied this from another piece of code in this class. I wasn't sure whether we should always set this property, even when it's false, but I thought we can leave this for now.